azure/scope/machine.go

```diff
@@ -220,6 +221,91 @@ func (m *MachineScope) InboundNatSpecs(portsInUse map[int32]struct{}) []azure.Re
 
 // NICSpecs returns the network interface specs.
 func (m *MachineScope) NICSpecs() []azure.ResourceSpecGetter {
+	nicSpecs := []azure.ResourceSpecGetter{}
+
+	if len(m.AzureMachine.Spec.NetworkInterfaces) < 1 {
+		return append(nicSpecs, getDefaultNetworkInterfaceSpec(m))
+	}
+
+	for i, n := range m.AzureMachine.Spec.NetworkInterfaces {
+		if n.ID != "" {
+			continue
+		}
+		spec := &networkinterfaces.NICSpec{
+			Name:                  azure.GenerateNICName(m.Name()) + "-" + strconv.Itoa(i),
+			ResourceGroup:         m.ResourceGroup(),
+			Location:              m.Location(),
+			SubscriptionID:        m.SubscriptionID(),
+			MachineName:           m.Name(),
+			VNetName:              m.Vnet().Name,
+			VNetResourceGroup:     m.Vnet().ResourceGroup,
+			AcceleratedNetworking: n.AcceleratedNetworking,
+			IPv6Enabled:           m.IsIPv6Enabled(),
+			EnableIPForwarding:    m.AzureMachine.Spec.EnableIPForwarding,
+			IPConfigs:             []networkinterfaces.IPConfig{},
+		}
+
+		spec.SubnetName = n.SubnetName
+
+		// Check for control plane interface setup on interface 0
+		if m.Role() == infrav1.ControlPlane && i == 0 {
+			spec.PublicLBName = m.OutboundLBName(m.Role())
+			spec.PublicLBAddressPoolName = m.OutboundPoolName(m.OutboundLBName(m.Role()))
```

Comment:
Maybe move this line with `spec.PublicLBAddressPoolName` into the `if` block below since it would be overridden in the `else` block.